netplan/cli/commands/status.py

```diff
@@ -0,0 +1,656 @@
+
+    def query_networkd(self) -> JSON:
+        data: JSON = None
+        try:
+            output: str = subprocess.check_output(['networkctl', '--json=short'],
+                                                  universal_newlines=True)
+            data = self.process_networkd(output)
+        except Exception as e:
+            logging.critical('Cannot query networkd interface data: {}'.format(str(e)))
+        return data
+
+    def process_nm(self, cmd_output) -> JSON:
+        data: JSON = []
+        for line in cmd_output.splitlines():
+            split = line.split(':')
+            dev = split[0] if split[0] else None
+            if dev:  # ignore inactive connection profiles
+                data.append({
+                    'device': dev,
+                    'name': split[1],
+                    'uuid': split[2],
+                    'filename': split[3],
+                    'type': split[4],
+                    'autoconnect': split[5],
+                    })
+        return data
+
+    def query_nm(self) -> JSON:
+        data: JSON = None
+        try:
+            output: str = subprocess.check_output(['nmcli', '-t', '-f',
+                                                   'DEVICE,NAME,UUID,FILENAME,TYPE,AUTOCONNECT',
+                                                   'con', 'show'],
+                                                  universal_newlines=True)
+            data = self.process_nm(output)
+        except Exception as e:
+            logging.debug('Cannot query NetworkManager interface data: {}'.format(str(e)))
+        return data
+
+    def query_routes(self) -> tuple:
+        data4 = None
+        data6 = None
+        try:
+            output4: str = subprocess.check_output(['ip', '-d', '-j', 'route'],
+                                                   universal_newlines=True)
+            data4: JSON = self.process_generic(output4)
+            output6: str = subprocess.check_output(['ip', '-d', '-j', '-6', 'route'],
+                                                   universal_newlines=True)
+            data6: JSON = self.process_generic(output6)
+        except Exception as e:
+            logging.debug('Cannot query iproute2 route data: {}'.format(str(e)))
+        return (data4, data6)
+
+    def query_resolved(self) -> tuple:
+        addresses = None
+        search = None
+        try:
+            ipc = dbus.SystemBus()
+            resolve1 = ipc.get_object('org.freedesktop.resolve1', '/org/freedesktop/resolve1')
+            resolve1_if = dbus.Interface(resolve1, 'org.freedesktop.DBus.Properties')
+            res = resolve1_if.GetAll('org.freedesktop.resolve1.Manager')
+            addresses = res['DNS']
+            search = res['Domains']
+        except Exception as e:
+            logging.debug('Cannot query resolved DNS data: {}'.format(str(e)))
+        return (addresses, search)
+
+    def pretty_print(self, data: JSON, total: int, _console_width=None) -> None:
```

Comment:
I've found by accident that sequences like `:ab:` and `:1234:` are being converted to emojis: 

```
● 13: br123 bridge DOWN/UP (NetworkManager: br123)
      MAC Address: ee:21:ad:24:f9:71
        Addresses: a🆎b🆎a🔢a:abcd/128
           Routes: a🆎b🆎a🔢a:abcd metric 426

```